specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/azureADOnlyAuthentication.json

```diff
@@ -0,0 +1,299 @@
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            },
+            "description": ""
+          }
+        },
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "Create or Update Azure Active Directory Only Authentication property": {
+            "$ref": "./examples/CreateOrUpdateAzureADOnlyAuthentication.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/azureADOnlyAuthentications": {
+      "get": {
+        "operationId": "AzureADOnlyAuthentications_List",
+        "summary": "Gets a list of Azure Active Directory only authentication property",
+        "description": "Gets a list of Azure Active Directory only authentication property for a workspace",
+        "tags": [
+          "WorkspaceAzureADOnlyAuthentications"
+        ],
+        "deprecated": false,
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./azureADOnlyAuthentication.json#/parameters/workspaceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "schema": {
+              "$ref": "#/definitions/AzureADOnlyAuthenticationListResult"
+            },
+            "description": "OK"
+          },
+          "default": {
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            },
+            "description": ""
+          }
+        },
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Get a list of Azure Active Directory Only Authentication property": {
+            "$ref": "./examples/ListAzureADOnlyAuthentication.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AzureADOnlyAuthenticationProperties": {
+      "description": "Properties of a active directory only authentication.",
+      "required": [
+        "azureADOnlyAuthentication"
+      ],
+      "type": "object",
+      "properties": {
+        "azureADOnlyAuthentication": {
```

Comment:
>azureADOnlyAuthentication [](http://example.com/codeflow?start=9&length=25)

ARM recommends string enum instead of boolean. This is better represented as authenticationType: [ "ActiveDirectoryOnly", "Default (or correct name)", <maybe others now or in future> ]. The enum values can actually be used to give information to the API users.

Reply:
We used boolean to maintain uniformity among similar controllers . For this we refer, SQL and SQL OD API documentation on configuring AzureADOnlyAuthentication, and they all were using boolean. So, we also implemented in a similar fashion.

SQL RP: https://docs.microsoft.com/en-us/rest/api/sql/2021-02-01-preview/server-azure-ad-only-authentications/create-or-update#creates-or-updates-azure-active-directory-only-authentication-object.